protocol/test/forge/v2/safeGuards/ThresholdSafeGuard.t.sol

```diff
@@ -277,165 +546,207 @@ contract ThresholdSafeGuardTest is ThresholdSafeGuardTestBase {
+            )
+        );
     }
 
     function test_checkTransaction_3Required_2Found() public {
         vm.startPrank(executor);
         guard.setDefaultSignaturesThreshold(3);
-        (uint256 pk2,) = addSafeOwners();
 
-        bytes32 dataHash = getTxHash("", 0);
-        bytes memory signature = bytes.concat(
-            signexecutor(safeOwners[0]), 
-            signEOA(dataHash, pk2)
+        checkTransaction(
+            ExpectSuccessOrFailure.FAILURE_THRESHOLD,
+            safeOwners[0], 
+            makeTransactionParams(
+                0, 
+                "", 
+                makeSignSpecs(
+                    SignSpec(SignType.AS_EXECUTOR, safeOwners[0]),
+                    SignSpec(SignType.AS_EOA_TX, safeOwners[1])
+                )
+            )
         );
-
-        vm.expectRevert("!Dynamic Signature Threshold");
-        doCheck("", 0, safeOwners[0], signature);
     }
 
     function test_checkTransaction_3Required_3Found() public {
         vm.startPrank(executor);
         guard.setDefaultSignaturesThreshold(3);
-        (uint256 pk2, uint256 pk3) = addSafeOwners();
 
-        bytes32 dataHash = getTxHash("", 0);        
-        bytes memory signature = bytes.concat(
-            signexecutor(safeOwners[0]), 
-            signEOA(dataHash, pk2), 
-            signEOA(dataHash, pk3)
+        checkTransaction(
+            ExpectSuccessOrFailure.SUCCESS,
+            safeOwners[0], 
+            makeTransactionParams(
+                0, 
+                "", 
+                makeSignSpecs(
+                    SignSpec(SignType.AS_EXECUTOR, safeOwners[0]),
+                    SignSpec(SignType.AS_EOA_TX, safeOwners[1]),
+                    SignSpec(SignType.AS_EOA_TX, safeOwners[2])
+                )
+            )
         );
-
-        doCheck("", 0, safeOwners[0], signature);
     }
 
     function test_checkTransaction_badPreApproval() public {
         vm.startPrank(executor);
         guard.setDefaultSignaturesThreshold(3);
-        (uint256 pk2, uint256 pk3) = addSafeOwners();
-
-        bytes32 dataHash = getTxHash("", 0);        
-        bytes memory signature = bytes.concat(
-            signexecutor(safeOwners[0]), 
-            signEOA(dataHash, pk2), 
-            signEOA(dataHash, pk3)
-        );
 
         // Check with a different executor to what we signed with
-        vm.expectRevert("GS025");
-        doCheck("", 0, safeOwners[1], signature);
+        checkTransaction(
+            ExpectSuccessOrFailure.FAILURE_INCORRECT_EXECUTOR,
+            safeOwners[1], // Not the same as the AS_EXECUTOR below
+            makeTransactionParams(
+                0, 
+                "", 
+                makeSignSpecs(
+                    SignSpec(SignType.AS_EXECUTOR, safeOwners[0]),
+                    SignSpec(SignType.AS_EOA_TX, safeOwners[1]),
+                    SignSpec(SignType.AS_EOA_TX, safeOwners[2])
+                )
```

Comment:
for curiosity, could you create a test with duplicate signers

Reply:
The sig check ensures that each signer address is always > than the last 

https://github.com/safe-global/safe-contracts/blob/186a21a74b327f17fc41217a927dea7064f74604/contracts/GnosisSafe.sol#L301

This also means the sigs need to be sorted in order of address too, which I've done by construction rather than add a quicksort to the solidity test.

Added the test in https://github.com/TempleDAO/temple/pull/842/commits/c15b3684c8653f0ef6804836d08de11920e85f81